File: ios/PanicKeywordDetector.swift
//
//  PanicKeywordDetector.swift
//  Protecta
//
//  Created by GitHub Copilot
//

import Foundation

class PanicKeywordDetector {
    private static let panicKeywords = [
        "help",
        "save me",
        "emergency",
        "danger",
        "i am unsafe"
    ]

    static func isPanicKeyword(_ speechText: String) -> Bool {
        let normalized = speechText.lowercased()
        return panicKeywords.contains { normalized.contains($0) }
    }

    static func detectKeyword(_ speechText: String) -> String? {
        let normalized = speechText.lowercased()
        return panicKeywords.first { normalized.contains($0) }
    }
}

File: app/src/main/java/com/example/safeher/utils/PanicKeywordDetector.java
package com.example.safeher.utils;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;

public class PanicKeywordDetector {

    private static final List<String> PANIC_KEYWORDS = Arrays.asList(
            "help",
            "save me",
            "emergency",
            "danger",
            "i am unsafe"
    );

    private PanicKeywordDetector() {
    }

    public static String detectKeyword(String speechText) {
        if (speechText == null) {
            return null;
        }

        String normalized = speechText.toLowerCase(Locale.US);
        for (String keyword : PANIC_KEYWORDS) {
            if (normalized.contains(keyword)) {
                return keyword;
            }
        }
        return null;
    }
}
